bionmr_utils/md/process/select.py

```diff
@@ -19,67 +31,60 @@ def get_methyl_vectors(frame: Frame) -> List[Tuple[Atom, Atom]]:
                 ('ILE', 'CG2'): ['CG2', 'HG21'],
                 ('MET', 'CE1'): ['CE1', 'HE1']}
 
-    atom_pairs = []
-
-    for r in frame.asResidues:
-        for atom in r.asAtoms:
-            if (r.name.str, atom.name.str) in CH3_dict.keys():
-                C = r[AtomName(CH3_dict[(r.name.str, atom.name.str)][0])]
-                H = r[AtomName(CH3_dict[(r.name.str, atom.name.str)][1])]
-                atom_pairs.append((C, H))
-
-    return atom_pairs
+    C_ids = []
+    H_ids = []
 
+    for r in frame.residues:
+        for atom in r.atoms:
+            if (r.name, atom.name) in CH3_dict.keys():
+                C_ids.append(r[CH3_dict[(r.name, atom.name)][0]].id)
+                H_ids.append(r[CH3_dict[(r.name, atom.name)][1]].id)
 
-def get_NH_vectors(frame: Frame) -> List[Tuple[Atom, Atom]]:
-    """
+    C_atoms = frame.atoms.filter(aId.is_in(set(C_ids)))
+    H_atoms = frame.atoms.filter(aId.is_in(set(H_ids)))
+    atom_pairs_selection = (C_atoms, H_atoms)

```

Comment:
If I get it right the idea was to remove python loop over pairs of atoms and use "vectorized" operation, right? So the interface changed from list of pairs to pair of selections. Ok. 

The function reorder scrambled the diff, so it's impractical to write "github suggestions", so here it is as python code: 

```python
    C_ats = []
    H_ats = []

    for r in frame.residues:
        for atom in r.atoms:
            if (r.name, atom.name) in CH3_dict.keys():
                c_name, h_name = CH3_dict[r.name, atom.name]
                C_ats.append(r[c_name])
                H_ats.append(r[h_name])

    return AtomSelection(C_ats), AtomSelection(H_ats)
```